include/boost/unordered/detail/foa/table.hpp

```diff
@@ -128,10 +133,21 @@ class table_iterator
   template<typename> friend class table_erase_return_type;
   template<typename,typename,typename,typename> friend class table;
 
-  table_iterator(Group* pg,std::size_t n,const table_element_type* p_):
-    pc{reinterpret_cast<unsigned char*>(const_cast<group_type*>(pg))+n},
-    p{const_cast<table_element_type*>(p_)}
-    {}
+  table_iterator(group_type* pg,std::size_t n,const table_element_type* p_)
+  {
+    if(pg)
+    {
+      using pointer_traits=boost::pointer_traits<char_pointer>;
+      pc=pointer_traits::pointer_to(
+        reinterpret_cast<unsigned char*>(const_cast<group_type*>(pg))[n]);
+    }
+
+    if(p_)
```

Comment:
Same as above. You may want to assert non-nullness of both `pg` and `p_`.

Reply:
Ah, those assertions would fail. We are indeed constructing table_iterator's with null pointers, even in the develop branch.

Reply:
Yes, my bad assumption, this constructor can legitimately take null pointers.